### PR TITLE
Add phone_number field to shipping information (or, add buyer_phone_number field to order)

### PR DESCRIPTION
**Pull request by ds300**
Tested locally:
image

https://artsyproduct.atlassian.net/browse/PURCHASE-395

### CODE REVIEW

**Comment by ds300** [comment]
Currently we will only ask for the phone number for the SHIP fulfilment type. At least that's what it says in the Jira ticket (which I forgot to attach above, apologies) https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=41&projectKey=PURCHASE&modal=detail&selectedIssue=PURCHASE-394

Your suggestion makes sense as future-proofing though. I like it :+1: and will do that.

**Comment by ds300** [comment]
@ashkan18 updated 🍏 

image

File: app/graphql/mutations/set_shipping.rb [hunk 1]
```diff
@@ -3,16 +3,20 @@ class Mutations::SetShipping < Mutations::BaseMutation
...
   argument :shipping, Inputs::ShippingAttributes, required: false
 
   field :order_or_error, Mutations::OrderOrFailureUnionType, 'A union of success/failure', null: false
 
-  def resolve(id:, fulfillment_type:, shipping: {})
+  def resolve(id:, fulfillment_type:, phone_number:, shipping: {})
     order = Order.find(id)
     validate_buyer_request!(order)
+
+    raise Errors::AuthError, 'Phone number is required' if fulfillment_type == Order::SHIP && phone_number.nil?
```
Comment (ashkan18) [inline app/graphql/mutations/set_shipping.rb]: part of me wants to say lets make `phone_number` required all the time if its mandatory for shipping. I can imagine `pickup` is actually dependent on it! but we can do that in a separate PR.

**Comment by ashkan18** [comment]
As a side note, phone number is PII along with few other fields in Exchange, I've talked to @madeleineb about it and it seems for now we are good with keeping this data in Exchange since there is a good business reason behind it, but just something to keep in mind.

**Pull request merged**